mono/tests/runtime_test.cs

```diff
@@ -0,0 +1,948 @@
+// using System.Threading.Tasks;
+
+#if !FULL_AOT_DESKTOP && !MOBILE
+using Mono.Unix.Native;
+#endif
+
+using NUnit.Framework;
+
+namespace MonoTests
+{
+	[TestFixture]
+	public class Runtime
+	{
+		static bool Verbose =
+			Environment.GetEnvironmentVariable ("V") != null;
+
+		static string[] Disabled =
+			Environment.GetEnvironmentVariable ("MONO_TESTS_DISABLED")?
+					.Split (new [] { ' ' }, StringSplitOptions.RemoveEmptyEntries)
+				?? Array.Empty<string> ();
+
+		static Process CurrentProcess = Process.GetCurrentProcess ();
+
+		[TestFixtureSetUp]
+		public void TestFixtureSetUp ()
+		{
+			IEnumerable<string> tests =
+				TestsRaw.Concat (StressTestsRaw)
+					.Select (tcd => (string) tcd.Arguments [0])
+					.Where (filename => !Disabled.Contains (filename))
+					.Distinct ();
+
+			if (!tests.Any ())
+				return;
+
+			StringBuilder sb = new StringBuilder ();
+
+			sb.Append ("-j");
+			sb.Append (Environment.ProcessorCount);
+
+			foreach (string test in tests) {
+				sb.Append (" \"");
+				sb.Append (test);
+				sb.Append ("\"");
+			}
+
+			string arguments = sb.ToString ();
+
+			ProcessStartInfo psi = new ProcessStartInfo {
+				FileName = "make",
+				Arguments = arguments,
+				UseShellExecute = false,
+				RedirectStandardOutput = true,
+				RedirectStandardError = true,
+				CreateNoWindow = true,
+			};
+
+			using (Process process = new Process { StartInfo = psi }) {
+				if (Verbose) {
+					process.OutputDataReceived += delegate (object sender, DataReceivedEventArgs e)
+					{
+						if (e.Data != null)
+							Console.Out.WriteLine (e.Data);
+					};
+
+					process.ErrorDataReceived += delegate (object sender, DataReceivedEventArgs e)
+					{
+						if (e.Data != null)
+							Console.Error.WriteLine (e.Data);
+					};
+				}
+
+				process.Start ();
+				process.BeginOutputReadLine ();
+				process.BeginErrorReadLine ();
+
+				process.WaitForExit ();
+			}
+		}
+
+		[Test, TestCaseSource ("Tests")]
+		public int Run (string filename, int timeout, EnvDictionary env, string flags)
+		{
+			return RunTest (filename, timeout, env, flags);
+		}
+
+		[Test, TestCaseSource ("StressTests"), Category ("Stress")]
+		public int StressRun (string filename, int timeout, EnvDictionary env, string flags)
+		{
+			return RunTest (filename, timeout, env, flags);
+		}
+
+		static int RunTest (string filename, int timeout, EnvDictionary env, string flags)
+		{
+			if (Disabled.Contains (filename))
+				Assert.Ignore ("disabled");
+
+			if (!File.Exists (filename))
+				throw new MissingTestException ();
+
+			ProcessStartInfo psi = new ProcessStartInfo {
+				FileName = CurrentProcess.MainModule.FileName,
+				Arguments = flags + " \"" + filename + "\"",
+				UseShellExecute = false,
+				RedirectStandardOutput = true,
+				RedirectStandardError = true,
+				CreateNoWindow = true,
+			};
+
+			if (env != null) {
+				foreach (KeyValuePair<string, string> kv in env)
+					psi.EnvironmentVariables.Add (kv.Key, kv.Value);
+			}
```

Comment:
So it seems this doesn't support a bunch of things that the test-runner.cs did, e.g. setting a custom runtime, setting MONO_PATH and MONO_CONFIG etc. Are those not needed for the tests you implemented right now?